frame/child-bounties/src/lib.rs

```diff
@@ -0,0 +1,877 @@
+			#[pallet::compact] child_bounty_id: BountyIndex,
+			beneficiary: <T::Lookup as StaticLookup>::Source,
+		) -> DispatchResult {
+			let signer = ensure_signed(origin)?;
+			let beneficiary = T::Lookup::lookup(beneficiary)?;
+
+			// Ensure parent bounty exists, and is active.
+			let (parent_curator, _) = Self::ensure_bounty_active(parent_bounty_id)?;
+
+			ChildBounties::<T>::try_mutate_exists(
+				parent_bounty_id,
+				child_bounty_id,
+				|maybe_child_bounty| -> DispatchResult {
+					let mut child_bounty =
+						maybe_child_bounty.as_mut().ok_or(BountiesError::<T>::InvalidIndex)?;
+
+					// Ensure child-bounty is in active state.
+					if let ChildBountyStatus::Active { ref curator } = child_bounty.status {
+						ensure!(
+							signer == *curator || signer == parent_curator,
+							BountiesError::<T>::RequireCurator,
+						);
+						// Move the child-bounty state to pending payout.
+						child_bounty.status = ChildBountyStatus::PendingPayout {
+							curator: signer,
+							beneficiary: beneficiary.clone(),
+							unlock_at: frame_system::Pallet::<T>::block_number() +
+								T::BountyDepositPayoutDelay::get(),
+						};
+						Ok(())
+					} else {
+						Err(BountiesError::<T>::UnexpectedStatus.into())
+					}
+				},
+			)?;
+
+			// Trigger the event Awarded
+			Self::deposit_event(Event::<T>::Awarded {
+				index: parent_bounty_id,
+				child_index: child_bounty_id,
+				beneficiary,
+			});
+
+			Ok(())
+		}
+
+		/// Claim the payout from an awarded child-bounty after payout delay.
+		///
+		/// The dispatch origin for this call may be any signed origin.
+		///
+		/// Call works independent of parent bounty state, No need for parent
+		/// bounty to be in active state.
+		///
+		/// The Beneficiary is paid out with agreed bounty value. Curator fee is
+		/// paid & curator deposit is unreserved.
+		///
+		/// Child-bounty must be in "PendingPayout" state, for processing the
+		/// call. And instance of child-bounty is removed from DB on successful
+		/// call completion.
+		///
+		/// - `parent_bounty_id`: Index of parent bounty.
+		/// - `child_bounty_id`: Index of child bounty.
+		#[pallet::weight(<T as Config>::WeightInfo::claim_child_bounty())]
+		pub fn claim_child_bounty(
+			origin: OriginFor<T>,
+			#[pallet::compact] parent_bounty_id: BountyIndex,
+			#[pallet::compact] child_bounty_id: BountyIndex,
+		) -> DispatchResult {
+			let _ = ensure_signed(origin)?;
+
+			// Ensure child-bounty is in expected state.
+			ChildBounties::<T>::try_mutate_exists(
+				parent_bounty_id,
+				child_bounty_id,
+				|maybe_child_bounty| -> DispatchResult {
+					let child_bounty =
+						maybe_child_bounty.as_mut().ok_or(BountiesError::<T>::InvalidIndex)?;
+
+					if let ChildBountyStatus::PendingPayout {
+						ref curator,
+						ref beneficiary,
+						ref unlock_at,
+					} = child_bounty.status
+					{
+						// Ensure block number is elapsed for
+						// processing the claim.
```

Comment:
```suggestion
						// Ensure block number is elapsed for processing the claim.
```

Okay I couldn't resist not fixing this one :D